zebra-test/src/command.rs

```diff
@@ -3,33 +3,40 @@ use color_eyre::{
     Help, SectionExt,
 };
 use std::process::{Child, Command, ExitStatus, Output};
-use std::{fs, io::Write};
+use std::{fs, io::Write, path::PathBuf};
 use tempdir::TempDir;
 
 #[cfg(unix)]
 use std::os::unix::process::ExitStatusExt;
 
-/// Runs a command in a TempDir
-pub fn test_cmd(path: &str) -> Result<(Command, impl Drop)> {
-    let dir = TempDir::new(path)?;
-    let mut cmd = Command::new(path);
-    cmd.current_dir(dir.path());
-
-    let cache_dir = dir.path().join("state");
-    fs::create_dir(&cache_dir)?;
-
-    fs::File::create(dir.path().join("zebrad.toml"))?.write_all(
-        format!(
-            "[state]\ncache_dir = '{}'",
-            cache_dir
-                .into_os_string()
-                .into_string()
-                .map_err(|_| eyre!("tmp dir path cannot be encoded as UTF8"))?
-        )
-        .as_bytes(),
-    )?;
-
-    Ok((cmd, dir))
+/// Runs a command
+pub fn test_cmd(command_path: &str) -> Result<Command> {
+    let mut cmd = Command::new(command_path);
+    cmd.current_dir(PathBuf::from(command_path).parent().unwrap().to_path_buf());
```

Comment:
Instead of setting the current dir to the directory storing the binary we're testing we should probably pass in the path for the tempdir.

Reply:
https://github.com/ZcashFoundation/zebra/pull/859/commits/a27e5a243fb98084cf6ee8ce7852113661e5a9c5